apps/theming-designer/src/components/FabricSlotWidget.tsx

```diff
@@ -0,0 +1,100 @@
+import * as React from 'react';
+import { IThemeRules, FabricSlots, IThemeSlotRule } from 'office-ui-fabric-react/lib/ThemeGenerator';
+import { IColor } from 'office-ui-fabric-react/lib/Color';
+import { Stack } from 'office-ui-fabric-react/lib/Stack';
+import { mergeStyles } from '@uifabric/merge-styles';
+import { ColorPicker } from 'office-ui-fabric-react/lib/ColorPicker';
+import { Callout, DirectionalHint } from 'office-ui-fabric-react/lib/Callout';
+import { Text } from 'office-ui-fabric-react';
+
+export interface IFabricSlotWidgetProps {
+  slot: FabricSlots;
+  onFabricPaletteColorChange: (newColor: IColor | undefined, fabricSlot: FabricSlots) => void;
+  themeRules?: IThemeRules;
+  directionalHint?: DirectionalHint;
+}
+
+export interface IFabricSlotWidgetState {
+  isColorPickerVisible: boolean;
+  colorPickerElement: HTMLElement | null;
+}
+
+const slotClassName = mergeStyles({
+  display: 'flex',
+  alignItems: 'center',
+  overflow: 'auto'
+});
+
+const fabricPaletteColorBox = mergeStyles({
+  width: 15,
+  height: 15,
+  display: 'inline-block',
+  left: 5,
+  top: 5,
+  border: '1px solid black',
+  flexShrink: 0
+});
+
+const colorBoxAndHexStringClassName = mergeStyles({
+  position: 'relative' /* This is necessary to make position: absolute; work in the other style. */
+});
+
+export class FabricSlotWidget extends React.Component<IFabricSlotWidgetProps, IFabricSlotWidgetState> {
+  constructor(props: any) {
+    super(props);
+
+    this.state = {
+      isColorPickerVisible: false,
+      colorPickerElement: null
+    };
+
+    this._onColorBoxClick = this._onColorBoxClick.bind(this);
+    this._onCalloutDismiss = this._onCalloutDismiss.bind(this);
+  }
+
+  public render() {
+    const { isColorPickerVisible, colorPickerElement } = this.state;
+    const slotRule = this.props.themeRules![FabricSlots[this.props.slot]];
```

Comment:
Can we avoid these non-null assertions throughout the PR? They make logic difficult to reason about and defeat the purposes of type-safety. Perhaps we can verify the object is non-null.